probe-rs/src/vendor/microchip/mod.rs

```diff
@@ -28,4 +33,46 @@ impl Vendor for Microchip {
 
         Some(sequence)
     }
+
+    fn try_detect_arm_chip(
+        &self,
+        interface: &mut dyn ArmProbeInterface,
+        chip_info: ArmChipInfo,
+    ) -> Result<Option<String>, Error> {
+        if chip_info.manufacturer.get() != Some("Atmel") || chip_info.part != 0xCD0 {
+            return Ok(None);
+        }
+
+        // FIXME: This is a bit shaky but good enough for now.
+        let access_port = MemoryAp::new(ApAddress::with_default_dp(0));
```

Comment:
Why don't we just put this in the config while we are at it? :)

Reply:
The AP should instead be passed in from the caller. The detection should probably run using the same AP we used to read the ROM table. But that code needs way more refactoring than I wanted here, as it searches for a valid memory AP, but then we lose the info.